static/app/utils/duration/formatDuration.tsx

```diff
@@ -0,0 +1,114 @@
+import {formatSecondsToClock} from 'sentry/utils/formatters';
+
+type Unit = 'ms' | 'sec' | 'min' | 'hour' | 'day' | 'week';
+type Format =
+  // example: `3,600`
+  | 'count-locale'
+  // example: `24h`
+  | 'count-unit'
+  // example: `86400`
+  | 'count'
+  // example: `1:00:00.000`
+  | 'h:mm:ss.sss'
+  // example: `1:00:00`
+  | 'h:mm:ss'
+  // example: `01:00:00.000
+  | 'hh:mm:ss.sss'
+  // example: `01:00:00`
+  | 'hh:mm:ss';
+
+type Args = {
+  /**
+   * The precision of the output.
+   *
+   * If the output precision is more granular than the input precision you might
+   * find the output value is rounded down, because least-significant digits are
+   * simply chopped off.
+   * Alternativly, because of IEEE 754, converting from a granular precision to
+   * something less granular might, in some cases, change the least-significant
+   * digits of the final value.
+   */
+  precision: Unit;
+  /**
+   * The output style to use
+   *
+   * ie: 120 seconds formatted as "h:mm" results in "2:00"
+   * ie: 10500 formatted as "count" + "sec" results in "10.5"
+   */
+  style: Format;
+  /**
+   * The timespan/duration to be displayed
+   * ie: "1000 miliseconds" would have the same output as "1 second"
+   *
+   * If it's coming from javascript `new Date` then 'ms'
+   * If it's from an SDK event, probably 'sec'
+   */
+  timespan: [number, Unit];
```

Comment:
🚲 🏠 why a tuple instead of 2 args?

Reply:
I think to kind of glue them together. If it were exported like `export type Timespan = [number, Unit];` and that would be a thing to pass around.